lib/fileutils.c

```diff
@@ -246,3 +247,21 @@ char *stripoff_last_component(char *path)
 	*p = '\0';
 	return p + 1;
 }
+
+/* Copies the contents of a file. Returns -1 on read error, -2 on write error. */
+int ul_copy_file(int from, int to)
+{
+	ssize_t nr, nw, off;
+	char buf[8 * 1024];
+
+	while ((nr = read(from, buf, sizeof(buf))) > 0)
+		for (off = 0; nr > 0; nr -= nw, off += nw)
+			if ((nw = write(to, buf + off, nr)) < 0)
+				return -2;
```

Comment:
See include/all-io.h., it would be probably nice to use read_all() and write_all() here to be robust againts signals.